src/main/java/com/upserve/uppend/lookup/LongLookup.java

```diff
@@ -212,20 +213,26 @@ public void close() {
 
     @Override
     public void flush() {
+        if (log.isTraceEnabled()) {
+            log.trace("flushing {}", dir);
+        }
+        ConcurrentHashMap<Path, LookupData> cacheCopy;
         synchronized (writeCache) {
-            if (log.isTraceEnabled()) {
-                log.trace("flushing {} (~{} entries)", dir, writeCache.size());
-            }
-            writeCache.forEach((path, data) -> {
-                log.trace("cache flushing {}", path);
+            cacheCopy = new ConcurrentHashMap<>(writeCache);
+        }
+        ArrayList<Future> futures = new ArrayList<>();
+        cacheCopy.forEach((path, data) -> {
+            futures.add(AutoFlusher.flushExecPool.submit(() -> {
```

Comment:
I wonder if the thread pool should be in a third, centralized place, rather than in `AutoFlusher` and used here?

Reply:
Agree on this, but not sure what to do yet.